requests-for-comments/2023-03-01-Temporal-RFC.md

```diff
@@ -0,0 +1,239 @@
+# RFC: Temporal at PostHog
+
+Author: James Greenhill
+
+Date: February 27, 2023 
+
+# For the reader
+
+Hi there, dear reader. I would love for you to scrutinize this RFC and the following RFC with more specifics on how we can build out our workflows. I’m looking for two things in particular from you though:
+
+- Is this a good idea?
+- Is getting this done feasible?
+
+I’m specifically looking for feedback from teams that are directly impacted by this, namely:
+
+- Infrastructure
+- Billing
+- Pipeline
```

Comment:
Why? These are the teams that are directly going to have to deal with the implications here.

I'd love your feedback as well, but you shouldn't have to live with the day to day implications here. I'm just making sure those teams take a look and understand how this may effect them.

Reply:
Well, scheduling long running tasks is something we're doing all the time in product analytics. Be it simple things like insight or dashboard refreshes, long running export queries, calculations to get usage counts for events or properties, you name it. I'm not even touching cohorts, which are technically not part of team product analytics.

I feel like there's a big divide simmering between app and platform, and the worlds should come closer together, not drift further apart. Hence, I wrote that I feel explicitly excluded.

Reply:
Ah, that's constructive feedback. Thank you.

Some direct feedback: 'I feel explicitly excluded' is not constructive feedback.
I would also argue that the comment does suggests 'bad intent' which is not a great place to start. I can tell you with authority that I did not explicitly exclude you.

I agree and hadn't considered that. You all should be added 💯 . These are great fits and your input is valuable as such!
- Cache refreshes (for insights and dashboards)
- Tasks for collecting statistics for Taxonomy

It's also a great point with cohorts. This RFC should include experimentation as well as they would be impacted by this.

I'm a bit confused as to why 'long running export queries' is something Product Analytics maintain? Could you explain that a bit? I agree it should fit into this just would like to understand who's owning it to makes sure they are also included.

>I feel like there's a big divide simmering between app and platform, and the worlds should come closer together, not drift further apart. Hence, I wrote that I feel explicitly excluded.

A huge reason why I have spent so much time on this RFC is to make sure that we work together closer. I'm trying to extend an olive branch out to every impacted team that this proposal effects. What would push us even further apart is for pipeline to ship this entirely in isolation, without the input from the effected teams, effectively alienating and isolating them from being able to provide feedback until it is too late.

Reply:
> I'm a bit confused as to why 'long running export queries' is something Product Analytics maintain?

Well, there's this button on the events table that runs through a script:

<img width="339" alt="image" src="https://user-images.githubusercontent.com/53387/222808747-129baa6e-60cf-4dbb-8056-d2f2c4f1b628.png">

And we're building Direct SQL access. I don't know what's your definition of "long running" in this case, but for me it's anything with timeouts > 60sec that get delivered async.

> What would push us even further apart is for pipeline to ship this entirely in isolation, without the input from the effected teams, effectively alienating and isolating them from being able to provide feedback until it is too late.

I've been feeling the same. I see all these problems, and a clear path on how to fix them. So I keep pushing out my own RFCs, giving brown bags, pushing ideas in meetings, etc... and then only getting negative and discouraging feedback in response. The feedback has always been ultimately useful and guided the direction in some way, but always phrased as "don't do this" instead of "tweak it a bit so".  Then I have nothing else to do but to build things in isolation, and watch the 360 feedback write itself :).